lib/matplotlib/backends/backend_pdf.py

```diff
@@ -2210,14 +2210,14 @@ def alpha_cmd(self, alpha, forced, effective_alphas):
         name = self.file.alphaState(effective_alphas)
         return [name, Op.setgstate]
 
-    def hatch_cmd(self, hatch):
+    def hatch_cmd(self, hatch, hatch_color):
```

Comment:
Because of the way the pdf backend deals with the fact that internally pdf's have a stack-based graphics context so when we change the gc in the draw methods, the `delta` method is used to determine what is different between the 'current' gc and the 'gc' that was just passed in.  The way that method works is by checking a bunch of internal state and if it needs to be updated emit some commands to the pdf file.  `hatch_cmd` is what generates the hatch commands.  To tell if we need to change the hatch we need to check both `_hatch` and `_hatch_color` and then all of the parameters you check are passed into the command function (see L2323 ish).  And the instance of the `gc` that the command methods are bound to is the 'old' gc, not the 'new' gc so we have to inject it this way.

It is possible that the pdf backend could be cleaned up, but that is a much bigger project.


sorry this is rambley, but it took me a while to sort this out last night.